http.go

```diff
@@ -83,40 +83,90 @@ type StdResponseWriter interface {
 	WriteString(data string) (int, error)
 }
 
-type logResponseWriter struct {
+// StdResponseWriter2 is the interface implemented by
+// the ResponseWriter from http.Server for HTTP/2 requests.
+//
+// HTTPServer's ResponseWriter implements this as well.
+type StdResponseWriter2 interface {
+	http.ResponseWriter
+	http.Flusher
+	http.CloseNotifier
+	http.Pusher
+	WriteString(data string) (int, error)
+}
+
+type logResponseWriter1 struct {
 	StdResponseWriter
 	status int
 	size   int64
 }
 
-func (w *logResponseWriter) WriteHeader(status int) {
+func (w *logResponseWriter1) WriteHeader(status int) {
 	w.status = status
 	w.StdResponseWriter.WriteHeader(status)
 }
 
-func (w *logResponseWriter) Write(data []byte) (int, error) {
+func (w *logResponseWriter1) Write(data []byte) (int, error) {
 	n, err := w.StdResponseWriter.Write(data)
 	w.size += int64(n)
 	return n, err
 }
 
-func (w *logResponseWriter) ReadFrom(r io.Reader) (int64, error) {
+func (w *logResponseWriter1) ReadFrom(r io.Reader) (int64, error) {
 	n, err := w.StdResponseWriter.ReadFrom(r)
 	w.size += n
 	return n, err
 }
 
-func (w *logResponseWriter) WriteString(data string) (int, error) {
+func (w *logResponseWriter1) WriteString(data string) (int, error) {
 	n, err := w.StdResponseWriter.WriteString(data)
 	w.size += int64(n)
 	return n, err
 }
 
+type logResponseWriter2 struct {
+	StdResponseWriter2
+	status int
+	size   int64
+}
+
+func (w *logResponseWriter2) WriteHeader(status int) {
+	w.status = status
+	w.StdResponseWriter2.WriteHeader(status)
+}
+
+func (w *logResponseWriter2) Write(data []byte) (int, error) {
+	n, err := w.StdResponseWriter2.Write(data)
+	w.size += int64(n)
+	return n, err
+}
+
+func (w *logResponseWriter2) WriteString(data string) (int, error) {
+	n, err := w.StdResponseWriter2.WriteString(data)
+	w.size += int64(n)
+	return n, err
+}
+
 // ServeHTTP implements http.Handler interface.
 func (s *HTTPServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	startTime := time.Now()
 
-	lw := &logResponseWriter{w.(StdResponseWriter), http.StatusOK, 0}
+	var rw http.ResponseWriter
+	var status *int
+	var size *int64
+	if srw1, ok := w.(StdResponseWriter); ok {
```

Comment:
The `ServeHTTP` function gets pretty long with this, so how about instead extracting this logic into a `createLogResponseWriter()` function? This would also be useful if for instance we need to add extra logic to support HTTP/3 in the future.

Reply:
Great suggestion.  Will do so.